### PR TITLE
Improve documentation regarding analyzers types

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

This pr adds the `plain` analyzer to the analyzer documentation and changes the description for the default analyzer for `text` columns from `standard` to `plain` in the fulltext indices documentation. It also adds missing ticks to the documentation of the `keyword` analyzer. 

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/general/ddl/analyzers.rst [hunk 1]
```diff
@@ -110,6 +110,15 @@ This is the same as the `standard-analyzer`_ analyzer.
...
 Uses the :ref:`lowercase-tokenizer` tokenizer.
 
+.. _plain-analyzer:
+
+``plain``
+----------
+
+``type='plain'``
+
+Creates one single token without any further analysis from the field-contents.
```
Comment (seut) [inline docs/general/ddl/analyzers.rst]:
```suggestion
This is the same as the `keyword`_ analyzer.
```

Comment (seut) [reply docs/general/ddl/analyzers.rst]: `plain` is a special CrateDB analyzer alias, I'm not sure if we allow to extend from it when creating custom analyzer.
Can you please verify that and note it here if not extendable (note to extend from `keyword`_ analyzer instead)?

Comment (mkleen) [reply docs/general/ddl/analyzers.rst]: Good point, thank you, will clarify that.

Comment (mkleen) [reply docs/general/ddl/analyzers.rst]: I am actually not quite sure if it would be better to drop the plain analyzer completely and simply reference for `plain` to the keyword analyzer.

File: docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -95,7 +95,9 @@ characters are allowed. Example::
     ... );
     CREATE OK, 1 row affected (... sec)
 
-Columns of type text can also be analyzed. See :ref:`sql_ddl_index_fulltext`.
+Columns of type text can also be analyzed. By default the
+:ref:`plain <plain-analyzer>` analyzer is used. See :ref:`sql_ddl_index_fulltext`
```
Comment (mechanomi) [inline docs/general/ddl/data-types.rst]: you can break lines inside of the ref blocks if there's a space inside the backticks

File: docs/general/ddl/data-types.rst [hunk 3]
```diff
@@ -761,7 +763,9 @@ the schema of the table. Once a column is added, it shows up in the
 ``information_schema.columns`` table and its type and attributes are fixed.
 They will have the type that was guessed by their inserted/updated value and
 they will always be ``not_indexed`` which means they are analyzed with the
-``plain`` analyzer, which means as-is.
+:ref:`plain <plain-analyzer>`, which means as-is.
+
+
```
Comment (mechanomi) [inline docs/general/ddl/data-types.rst]: extraneous?

File: docs/general/ddl/analyzers.rst [hunk 4]
```diff
@@ -110,6 +110,17 @@ This is the same as the `standard-analyzer`_ analyzer.
...
 Uses the :ref:`lowercase-tokenizer` tokenizer.
 
+.. _plain-analyzer:
+
+``plain``
+----------
+
+``type='plain'``
+
+This is the same as the `keyword`_ analyzer.
```
Comment (mechanomi) [inline docs/general/ddl/analyzers.rst]: can you rephrase these three lines as a single paragraph?

something like:

> The ``plain`` analyzer is an alias for the `keyword`_ analyzer and cannot be extended. You must extend the `keyword`_ analyzer instead.

Comment (mkleen) [reply docs/general/ddl/analyzers.rst]: 👍

File: docs/general/ddl/analyzers.rst (hunk 1 above)
Comment (mkleen) [reply docs/general/ddl/analyzers.rst]: @mechanomi what is your opinion here ?

Comment (mechanomi) [reply docs/general/ddl/analyzers.rst]: I think it makes sense to keep it in, and explain that it's an alias

Comment (mkleen) [reply docs/general/ddl/analyzers.rst]: 👍
